packages/dss-bridge/src/DomainHost.sol

```diff
@@ -0,0 +1,305 @@
+        daiJoin.join(address(vow), wad);
+
+        emit Surplus(wad);
+    }
+
+    /// @notice Cover the remote domain's deficit by pulling debt from the surplus buffer
+    function deficit(uint256 wad) external guestOnly vatLive {
+        vat.suck(vow, address(this), wad * RAY);
+        daiJoin.exit(address(escrow), wad);
+        
+        // Send ERC20 DAI to the remote DomainGuest
+        _rectify(wad);
+
+        emit Deficit(wad);
+    }
+
+    /// @notice Initiate shutdown for this domain
+    /// @dev This will trigger the end module on the remote domain
+    function cage() external {
+        require(vat.live() == 0 || wards[msg.sender] == 1, "DomainHost/not-authorized");
+        require(live == 1, "DomainHost/not-live");
+
+        live = 0;
+
+        _cage();
+
+        emit Cage();
+    }
+
+    /// @notice Set this domain's cure value
+    function tell(uint256 value) external guestOnly {
+        require(live == 0, "DomainHost/live");
+        require(!cureReported, "DomainHost/cure-reported");
+        require(_divup(value, RAY) <= grain, "DomainHost/cure-bad-value");
+
+        cureReported = true;
+        cure = value;
+
+        emit Tell(value);
+    }
+
+    /// @notice Allow DAI holders to exit during global settlement
+    /// @dev    This will mint a pro-rata claim token on the remote domain.
+    ///         Gem amount is scaled by the actual debt of the remote domain.
+    ///         `usr` is the address for the mint on the remote domain.
+    function exit(address usr, uint256 wad) external {
+        require(vat.live() == 0, "DomainHost/vat-live");
+        vat.slip(ilk, msg.sender, -_int256(wad));
+
+        // Convert to actual debt amount
+        // Round against the user
+        uint256 claim = wad * (grain - _divup(cure, RAY)) / grain;
+        
+        _mintClaim(usr, claim);
+
+        emit Exit(usr, wad, claim);
+    }
+
+    // --- Canonical DAI Support ---
+
+    /// @notice Deposit local DAI to mint remote canonical DAI
+    /// @param to The address to send the DAI to on the remote domain
+    /// @param amount The amount of DAI to deposit [WAD]
+    function deposit(address to, uint256 amount) external {
+        require(dai.transferFrom(msg.sender, escrow, amount), "DomainHost/transfer-failed");
+
+        _deposit(to, amount);
+
+        emit Deposit(to, amount);
+    }
+
+    /// @notice Withdraw local DAI by burning remote canonical DAI
+    /// @param to The address to send the DAI to on the local domain
+    /// @param amount The amount of DAI to withdraw [WAD]
+    function withdraw(address to, uint256 amount) external guestOnly {
+        require(dai.transferFrom(escrow, to, amount), "DomainHost/transfer-failed");
+
+        emit Withdraw(to, amount);
+    }
+
+    // --- Maker Teleport Support ---
+
+    /// @notice Finalize a teleport registration
+    function finalizeTeleport(TeleportGUID calldata teleport) external guestOnly {
```

Comment:
Should this be called something like `teleportViaSlowPath` ? In general we used the `finalize` prefix in Teleport Gateways for functions called via a cross chain message, but here we don't use that convention for other methods.